tools/mmp/driver.cs

```diff
@@ -614,6 +614,18 @@ static void Pack (IList<string> unprocessed)
 					else
 						native_libs.Add (kvp.Key, kvp.Value);
 				}
+				// Idealy, this would be handled by Link () above. However in the non-linking case
+				// we do not run MobileMarkStep which generates the pinvoke list. Hack around this for now
+				// https://bugzilla.xamarin.com/show_bug.cgi?id=43419
+				if (App.LinkMode == LinkMode.None) {
+					foreach (var kvp in ProcessDllImports ()) {
```

Comment:
to reduce code duplication (loop) could you not just have `Link` return `ProcessDllImports` if `LinkMode` == `None` ? otherwise it seems fine (but the code could diverge and introduce bugs)